### PR TITLE
Theme preview implementation using theme providers

### PR DESCRIPTION
**Pull request by alex-mtl**
Relates to: https://github.com/vanilla/knowledge/issues/1226

### CODE REVIEW

File: library/src/scripts/features/toaster/themePreview/ThemePreviewToastReducer.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import { ICoreStoreState } from "@library/redux/reducerRegistry";
+
+export interface IPreviewToasterState {
+    applyStatus: ILoadable<{ themeID: number | string }>;
+    cancelStatus: ILoadable<{ themeID: number | string }>;
+}
+
+export interface IPreviewToasterStoreState extends ICoreStoreState {
+    themePreviewToaster: IPreviewToasterState;
+}
```
Comment (charrondev) [inline library/src/scripts/features/toaster/themePreview/ThemePreviewToastReducer.ts]: Could we rename this file and move it up into `/theming`? I'm thinking `themePreviewReducer`

Nothing here is really specific to the toast itself.

File: library/src/scripts/theming/ThemesActions.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/**
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license Proprietary
+ */
```
Comment (charrondev) [inline library/src/scripts/theming/ThemesActions.ts]: Incorrect license for this repo.

File: library/src/scripts/theming/ThemesActions.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    );
+
+    public static readonly putCurrentThemeACs = actionCreator.async<
+        { themeID: number | string },
+        IManageTheme,
+        IApiError
+    >("PUT_CURRENT");
+
+    public static readonly putPreviewThemeACs = actionCreator.async<IPutCurrentThemeRequest, IManageTheme, IApiError>(
+        "PUT_PREVIEW",
```
Comment (charrondev) [inline library/src/scripts/theming/ThemesActions.ts]: We have a `ThemeActions` right next to this one. It seems really confusing to have both. Could we merge this into that one?

**Comment by alex-mtl** [comment]
> This seems to be missing some reducer registrations in order to function without the new plugin. Essentially your stuff from the "common" entry in the API plugin probably needs to be moved up here.

I've tested on my localhost with plugin disabled. It works to me. Do you see something missing in particular?

**Comment by slafleche** [comment]
I can't get this to run. Even with both branches up to date, i get an error.

**Comment by charrondev** [comment]
The issue @slafleche had should be resolved by the merge to master that was done here.

**Pull request merged**